lib/taxonomy/publishing_api_linked_edition_parser.rb

```diff
@@ -0,0 +1,38 @@
+module Taxonomy
+  class PublishingApiLinkedEditionParser
+    attr_accessor :linked_edition
+
+    def initialize(edition_response)
+      @linked_edition = Taxon.new(edition_response.to_h.symbolize_keys)
+      @name_field = name_field
+    end
+
+    def add_expanded_links(expanded_links_response)
```

Comment:
There is some duplication between `add_expanded_links` and `parse_nested_item` we could refactor maybe? Possibly by adding some recursion?

- Both look for a `child_taxons` key, but from either `expanded_links_response["expanded_links"]` or `nested_item["links"]`
- Both add a to the `children` collection, but either `linked_edition` or `nested_linked_edition`.